lib/scala/kg/src/main/scala/stores/kg/MemKgStore.scala

```diff
@@ -97,10 +97,10 @@ class MemKgStore extends KgStore {
   final override def putNodes(nodes: Iterator[KgNode]): Unit = {
     this.nodes = nodes.toList
     this.nodesById = this.nodes.map(node => (node.id, node)).toMap
-    this.datasources = this.nodes.flatMap(_.datasource.split(",")).distinct
+    this.sources = this.nodes.flatMap(_.sources).distinct
     lucene.deleteAll()
     this.nodes.foreach(node => {
-      lucene.doc().fields(luceneNodeDatasourceField(node.datasource), luceneNodeIdField(node.id), luceneNodeLabelField(node.label)).index()
+      lucene.doc().fields(luceneNodeSourcesField(node.sources.mkString(ListDelim)), luceneNodeIdField(node.id), luceneNodeLabelsField(node.labels.mkString(ListDelim))).index()
```

Comment:
Fields can have multiple values in Lucene. That's better than inserting a single value with delimiters in it.